pkgs/by-name/pr/proton-ge-custom/package.nix

```diff
@@ -0,0 +1,41 @@
+{ lib
+, stdenvNoCC
+, fetchurl
+, writeScript
+}:
+stdenvNoCC.mkDerivation (finalAttrs: {
+  pname = "proton-ge-custom";
+  version = "GE-Proton9-1";
+
+  src = fetchurl {
+    url = "https://github.com/GloriousEggroll/proton-ge-custom/releases/download/${finalAttrs.version}/${finalAttrs.version}.tar.gz";
+    hash = "sha256-wCIffeayOy3kEwmIKB7e+NrliuSpKXoVYC334fxVB3U=";
+  };
+
+  buildCommand = ''
+    runHook preBuild
+    mkdir -p $out/{bin,opt}
+    tar -C $out/opt --strip=1 -x -f $src
+    ln -s $out/opt/toolmanifest.vdf $out/bin/toolmanifest.vdf
+    install -Dm644 $out/opt/compatibilitytool.vdf $out/bin/compatibilitytool.vdf
+    sed -i "s#\"install_path\" \".\"#\"install_path\" \"$out\/opt\/\"#g"  $out/bin/compatibilitytool.vdf
+    runHook postBuild
+  '';
+
+  passthru.updateScript = writeScript "update-proton-ge" ''
+    #!/usr/bin/env nix-shell
+    #!nix-shell -i bash -p curl jq common-updater-scripts
+
+    repo="https://api.github.com/repos/GloriousEggroll/proton-ge-custom/releases"
+    version="$(curl -sL "$repo" | jq 'map(select(.prerelease == false)) | .[0].tag_name' --raw-output)"
+    update-source-version proton-ge-custom "$version"
+  '';
+
+  meta = {
```

Comment:
Should have sourceProvenance set since it's binary, not from source. I'd also prefer a `-bin` suffix for the attr name.

Reply:
I disagree with adding -bin suffix to the attr name. It makes sense, but `proton-ge-custom-bin` is an annoyingly long name for a package.

Reply:
On that note, what type should it be? I don't see a sourceeProvenance that fits our casee in https://github.com/NixOS/nixpkgs/blob/master/lib/source-types.nix

Reply:
There will likely be a regular from-source proton-ge at some point. Not having the suffix would put us in an annoying spot.

If anything, I'd drop the `custom` bit: `proton-ge-bin`.

`binaryNativeCode`.